app/src/screens/export-to-csv-flow/ExportToCsv.js

```diff
@@ -0,0 +1,243 @@
+import {
+  ActivityIndicator,
+  Alert,
+  PermissionsAndroid,
+  ScrollView,
+  Text,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import Share from 'react-native-share';
+import CompetitionsDB from '../../database/Competitions';
+import React, {useEffect, useState} from 'react';
+import {useTheme} from '@react-navigation/native';
+import NoInternet from '../../components/NoInternet';
+import ScoutReports from '../../database/ScoutReports';
+import FormsDB from '../../database/Forms';
+import {supabase} from '../../lib/supabase';
+import RNFS from 'react-native-fs';
+
+const ExportToCsv = () => {
+  const {colors} = useTheme();
+  const [internetError, setInternetError] = useState(false);
+  const [competitionList, setCompetitionList] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const getCompetitions = async () => {
+    try {
+      const data = await CompetitionsDB.getCompetitions();
+      // sort the data by start time
+      data.sort((a, b) => {
+        return new Date(a.startTime) - new Date(b.startTime);
+      });
+      setCompetitionList(data);
+      setInternetError(false);
+    } catch (error) {
+      console.error(error);
+      setInternetError(true);
+    }
+  };
+
+  useEffect(() => {
+    getCompetitions().catch(console.error);
+  }, []);
+
+  if (internetError) {
+    return <NoInternet colors={colors} onRefresh={getCompetitions()} />;
+  }
+
+  const exportToCsv = async comp => {
+    setLoading(true);
+    let form;
+    try {
+      form = await FormsDB.getForm(comp.formId);
+    } catch (error) {
+      console.error(error);
+      setLoading(false);
+      Alert.alert('Error', 'An error occurred while fetching the form');
```

Comment:
As a general thing for all alerts, we should give some info on how to resolve it. Maybe something along the lines of "An error occurred when fetching the form. Try reloading the app"
(this goes to all alerts, everywhere)

Reply:
👍